cmd/cmd.go

```diff
@@ -133,10 +150,169 @@ func initConfig() (config.Config, string, error) {
 }
 
 func initClient(logger logging.Logger, cfg config.Config) (pack.Client, error) {
-	client, err := pack.NewClient(pack.WithLogger(logger), pack.WithExperimental(cfg.Experimental), pack.WithRegistryMirrors(cfg.RegistryMirrors))
+	dc, err := initDockerClient()
+	if err != nil {
+		return pack.Client{}, err
+	}
+	client, err := pack.NewClient(pack.WithLogger(logger), pack.WithExperimental(cfg.Experimental), pack.WithRegistryMirrors(cfg.RegistryMirrors), pack.WithDockerClient(dc))
 	if err != nil {
 		return pack.Client{}, err
 	}
 
 	return *client, nil
 }
+
+func initDockerClient() (dockerClient.CommonAPIClient, error) {
+	dockerClientOpts := []dockerClient.Opt{
+		dockerClient.WithVersion("1.38"),
+	}
+
+	usesSSH := false
+	if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+		if _url, err := url.Parse(dockerHost); err == nil && _url.Scheme == "ssh" {
+			usesSSH = true
+			credentialsConfig := sshdialer.Config{
+				Identity:           os.Getenv("DOCKER_HOST_SSH_IDENTITY"),
+				PassPhrase:         os.Getenv("DOCKER_HOST_SSH_IDENTITY_PASSPHRASE"),
```

Comment:
How should we name these envvars?

Reply:
Are these the same ones that Docker uses? If so, we should keep them

Reply:
No they are not used by docker. It's my addition. With docker you need to put identity at default location with default name or add it to ssh-agent, you cannot specify identity file AFAIK.